test/extension/CMakeLists.txt

```diff
@@ -1,2 +1,10 @@
 build_loadable_extension_directory(loadable_extension_demo test/extension
                                    loadable_extension_demo.cpp)
+
+set(DUCKDB_TEST_ALIAS_UNITS test_alias_extension.cpp)
```

Comment:
As indicated before - can this use the same code path as the loadable extension demo (i.e. `build_loadable_extension_directory` and `LOAD`)? There are a number of code modifications that are not required here.